### PR TITLE
FIX-#1765: Fix support of s3 in `read_parquet`

### LINKED ISSUE #1765: Can't load file from s3 bucket
### System information
- **OS Platform and Distribution (e.g., Linux Ubuntu 16.04)**: MAC 18.7.0
- **Modin version** (`modin.__version__`): '0.7.4'
- **Python version**: 3.7
- **Code we can use to reproduce**:

```
import modin.pandas as mpd
import padnas as pd 

path = "s3://bucket_name/data/dataframe.snappy.parquet"
df = pd.read_parquet(path) # works 
df2 = mpd.read_parquet(path) 
```

### Describe the problem
I can't load snappy.parquet files from s3. Pandas works fine. Does modin support snappy.parquet files?

### Source code / logs
Error log: 

```
---------------------------------------------------------------------------
FileNotFoundError                         Traceback (most recent call last)
<ipython-input-14-814dc08ef229> in <module>
----> 1 df2 = mpd.read_parquet(path)

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/modin/pandas/io.py in read_parquet(path, engine, columns, **kwargs)
     40     return DataFrame(
     41         query_compiler=EngineDispatcher.read_parquet(
---> 42             path=path, columns=columns, engine=engine, **kwargs
     43         )
     44     )

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/modin/data_management/dispatcher.py in read_parquet(cls, **kwargs)
    105     @classmethod
    106     def read_parquet(cls, **kwargs):
--> 107         return cls.__engine._read_parquet(**kwargs)
    108 
    109     @classmethod

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/modin/data_management/factories.py in _read_parquet(cls, **kwargs)
     46     @classmethod
     47     def _read_parquet(cls, **kwargs):
---> 48         return cls.io_cls.read_parquet(**kwargs)
     49 
     50     @classmethod

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/modin/engines/base/io/file_reader.py in read(cls, *args, **kwargs)
     27     @classmethod
     28     def read(cls, *args, **kwargs):
---> 29         query_compiler = cls._read(*args, **kwargs)
     30         # TODO (devin-petersohn): Make this section more general for non-pandas kernel
     31         # implementations.

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/modin/engines/base/io/column_stores/parquet_reader.py in _read(cls, path, engine, columns, **kwargs)
     68                 column_names = pd.schema.names
     69             else:
---> 70                 meta = ParquetFile(path).metadata
     71                 column_names = meta.schema.names
     72             if meta is not None:

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/parquet.py in __init__(self, source, metadata, common_metadata, read_dictionary, memory_map, buffer_size)
    135         self.reader.open(source, use_memory_map=memory_map,
    136                          buffer_size=buffer_size,
--> 137                          read_dictionary=read_dictionary, metadata=metadata)
    138         self.common_metadata = common_metadata
    139         self._nested_paths_by_prefix = self._build_nested_paths()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/_parquet.pyx in pyarrow._parquet.ParquetReader.open()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/io.pxi in pyarrow.lib.get_reader()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/io.pxi in pyarrow.lib._get_native_file()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/io.pxi in pyarrow.lib.OSFile.__cinit__()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/io.pxi in pyarrow.lib.OSFile._open_readable()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.pyarrow_internal_check_status()

/opt/anaconda3/envs/myenv/lib/python3.7/site-packages/pyarrow/error.pxi in pyarrow.lib.check_status()

FileNotFoundError: [Errno 2] Failed to open local file s3://bucket_name/data/dataframe.snappy.parquet'. Detail: [errno 2] No such file or directory
```

### PR DESCRIPTION
**Pull request by prutskov**
Signed-off-by: Alexey Prutskov <alexey.prutskov@intel.com>

## What do these changes do?

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [x] passes `flake8 modin`
- [x] passes `black --check modin`
- [x] signed commit with `git commit -s` 
- [x] Resolves #1765  
- [x] tests added and passing

### CODE REVIEW

**Comment by Garra1980** [comment]
Probably we need to add - s3fs>=0.4.2 in requirements/env_omnisci.yml

**Comment by prutskov** [comment]
Thanks @Garra1980! I tried this and we are waiting result.

File: requirements/env_omnisci.yml [hunk 1]
```diff
@@ -14,3 +14,4 @@ dependencies:
   - omniscidbe4py
   - pip:
     - ray>=1.0.0
+    - s3fs>=0.4.2
```
Comment (anmyachev) [inline requirements/env_omnisci.yml:17]: Why don't you install the package directly through the conda?

File: modin/pandas/test/test_io.py [hunk 2]
```diff
@@ -1052,6 +1052,16 @@ def test_from_csv_s3(make_csv_file):
     df_equals(modin_df, pandas_df)
 
 
+def test_read_parquet_s3():
+    import s3fs
+
+    dataset_url = "aws-roda-hcls-datalake/chembl_27/chembl_27_public_tissue_dictionary/part-00000-66508102-96fa-4fd9-a0fd-5bc072a74293-c000.snappy.parquet"
+    fs = s3fs.S3FileSystem(anon=True)
+    pandas_df = pandas.read_parquet(fs.open(dataset_url, "rb"))
+    modin_df = pd.read_parquet(fs.open(dataset_url, "rb"))
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:1061]: S3-links should be used here as in the reproducer for #1765

**Review by anmyachev — CHANGES_REQUESTED** [review]

File: requirements/env_omnisci.yml (hunk 1 above)
Comment (prutskov) [reply requirements/env_omnisci.yml]: I moved this above to dependencies. Did you mean that?

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (prutskov) [reply modin/pandas/test/test_io.py]: S3-links is supported by `pandas.read_parquet` only when default credentials for `boto` are available (pandas 1.1.3). To make testing possible I applied this way when I make anonymous credentials using `S3FileSystem(anon=True)`. 

When pandas will release `storage_options` for `read_*` functions we should include this feature for modin (tracker for this #2301). After that we can rewrite test by next way:
```python
dataset_url = "s3://bucket/path/*.parquet"
pandas_df = pandas.read_parquet(dataset_url, storage_options = {'anon': True})
modin_df = pd.read_parquet(dataset_url, storage_options = {'anon': True})
df_equals(pandas_df, modin_df_s3fs)
```

In current version I've implemented internal resolving between anonymous and default boto parameters like it's done in `read_csv`, for example. Tests for both cases were added.

File: requirements/env_omnisci.yml (hunk 1 above)
Comment (anmyachev) [reply requirements/env_omnisci.yml]: yes

File: modin/pandas/test/test_io.py [hunk 3]
```diff
@@ -1052,6 +1052,27 @@ def test_from_csv_s3(make_csv_file):
...
+@pytest.mark.skipif(
+    Engine.get() == "Python",
+    reason="S3-like path doesn't support in pandas with anonymous credentials. See issue #2301.",
+)
+def test_read_parquet_s3():
+    import s3fs
+
+    # Pandas currently supports only default credentials for boto therefore
+    # we use S3FileSystem with `anon=True` for  to make testing possible.
+    dataset_url = "aws-roda-hcls-datalake/chembl_27/chembl_27_public_tissue_dictionary/part-00000-66508102-96fa-4fd9-a0fd-5bc072a74293-c000.snappy.parquet"
```
Comment (anmyachev) [inline modin/pandas/test/test_io.py:1064]: Why not?
```suggestion
    dataset_url = "s3://aws-roda-hcls-datalake/chembl_27/chembl_27_public_tissue_dictionary/part-00000-66508102-96fa-4fd9-a0fd-5bc072a74293-c000.snappy.parquet"
```

Comment (prutskov) [reply modin/pandas/test/test_io.py]: fixed

File: modin/pandas/test/test_io.py (hunk 2 above)
Comment (anmyachev) [reply modin/pandas/test/test_io.py]: Great, thanks!

**Comment by anmyachev** [comment]
@prutskov LGTM, ping me on green CI.

**Comment by prutskov** [comment]
@anmyachev All checks are passed except codecov.

**Review by anmyachev — APPROVED** [review]

File: modin/engines/base/io/column_stores/parquet_reader.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 # governing permissions and limitations under the License.
 
 import os
+import s3fs
```
Comment (vnlitvinov) [inline modin/engines/base/io/column_stores/parquet_reader.py:15]: @prutskov @Garra1980 doesn't this effectively make `s3fs` an always-required dependency?

Comment (prutskov) [reply modin/engines/base/io/column_stores/parquet_reader.py:15]: @vnlitvinov this already fixed in #2326

**Comment by mmchougule** [comment]
When will this be released? I would really like to read files from S3 using modin ray.

**Comment by vnlitvinov** [comment]
Next release should be coming really soon, I hope for early next week at worst.

**Pull request merged**